src/lib_json/json_value.cpp

```diff
@@ -182,6 +182,15 @@ Value::CZString::CZString(const CZString& other)
                      ? noDuplication : duplicate)
                  : other.index_) {}
 
+#ifdef JSON_HAS_RVALUE_REFERENCES
+Value::CZString::CZString(CZString&& other)
+	: cstr_(other.cstr_)
+	, index_(other.index_)
```

Comment:
This is my first contribution to github so I apologize if I've performed any faux pas :o)

As I replied to @BillyDonahue, I don't understand the binary compatibility concern, could you please enlighten me?

Of course I respect the fact that I'm contributing to an existing project and will respect the coding standards here.

You say that you want to pull and fix the merge, does this mean I should not make the suggested changes and submit a new pull request?